intact-orthology-import/src/main/java/uk/ac/ebi/intact/ortholog/OrthologsXrefWriter.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.ebi.intact.ortholog;
+
+import lombok.RequiredArgsConstructor;
+import psidev.psi.mi.jami.model.Xref;
+import uk.ac.ebi.intact.jami.dao.IntactDao;
+import uk.ac.ebi.intact.jami.model.extension.IntactCvTerm;
+import uk.ac.ebi.intact.jami.model.extension.IntactProtein;
+import uk.ac.ebi.intact.jami.model.extension.InteractorXref;
+import uk.ac.ebi.intact.jami.service.InteractorService;
+import uk.ac.ebi.intact.jami.utils.IntactUtils;
+import java.util.*;
+import java.util.Map;
+
+@RequiredArgsConstructor
+public class OrthologsXrefWriter {
+
+    private final static String PANTHER_DATABASE_ID = "EBI-968064";
+    private final static String PANTHER_DATABASE_MI = "MI:0702";
+    private final Map<String, IntactCvTerm> cvTermMap = new HashMap<>();
+    private final IntactDao intactDao;
+    private final InteractorService interactorService;
+
+
+    public void addOrthologyXref(IntactProtein protein, String pantherId) throws Exception{
+        InteractorXref xref = newOrthologsXref(pantherId);
+        if (!protein.getXrefs().contains(xref)){
+            protein.getXrefs().add(xref);
+        }
+        this.interactorService.saveOrUpdate(protein);
```

Comment:
We should not call `saveOrUpdate` here.

`OrthologsItemProcessor` process each protein, and add the xref. Then, `ProteinCollectionWriter` process a collection of updated proteins and saves them.

Currently, with this saveOrUpdate call here, we are saving the changes in the processor, as the processor calls this method; and then in the writer.

If you remove this saveOrUpdate call here, then it is still call in `ProteinCollectionWriter.write`, so changes are being saved.